utils/scripts/update_generated_code_from_idl.sh

```diff
@@ -3,9 +3,19 @@
 files_to_exclude=(
     './include/fastrtps/types/*'
     './include/fastdds/statistics/types.idl'
+    )
+
+files_needing_typeobject=(
+    './test/unittest/dynamic_types/idl/Basic.idl'
+    './test/unittest/dynamic_types/idl/new_features_4_2.idl'
+    './test/unittest/dynamic_types/idl/Test.idl'
```

Comment:
These other IDL files have also the TypeObject in their corresponding folder within the Fast DDS project. I am unsure if it is required or not. But I think that either we update the TypeObject or if the test does not require it, remove the unused files from the project.

* `test/unittest/xtypes/idl/Types.idl`
* `test/unittest/xtypes/idl/WideEnum.idl`
* `test/xtypes/idl/Types.idl`

Reply:
Done in 67797b9